config/config.go

```diff
@@ -0,0 +1,116 @@
+package config
+
+import (
+	_ "embed"
+	"errors"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/candrewlee14/webman/utils"
+
+	"gopkg.in/yaml.v3"
+)
+
+//go:embed default.yml
+var defaultConfig []byte
+
+type Config struct {
+	RefreshInterval time.Duration `yaml:"refresh_interval"`
+	PkgRepos        []PkgRepo     `yaml:"pkg_repos"`
+}
+
+type RefreshFile struct {
+	LastUpdated time.Time
+}
+
+type PkgRepoType string
+
+const (
+	PkgRepoTypeGitHub PkgRepoType = "github"
+	PkgRepoTypeGitea  PkgRepoType = "gitea"
+)
+
+type PkgRepo struct {
+	Name   string      `yaml:"name"`
+	Type   PkgRepoType `yaml:"type"`
+	User   string      `yaml:"user"`
+	Repo   string      `yaml:"repo"`
+	Branch string      `yaml:"branch"`
+
+	GiteaURL string `yaml:"gitea_url"`
```

Comment:
Should this be a `*string` to show that it's an optional field?

Reply:
Maybe a comment? YAML doesn't care, and the schema will be the guide when writing the actual YAML, so I didn't see a reason to make it harder to work with in the code.